trie/trie.go

```diff
@@ -73,36 +77,60 @@ func (t *Trie[V]) Put(key string, val V) {
 	if !t.Contains(key) {
 		t.n++
 	}
-	t.root = t.put(t.root, key, val, 0, true)
+	t.root = t.put(t.root, key, val, 0)
+}
+
+func (t *Trie[V]) put(x *node[V], key string, val V, d int) *node[V] {
```

Comment:
I moved `put` up to sit underneath `Put` rather than under `Remove`. The only functional change is the removal of the `valid` boolen parameter.